govcd/adminvdc.go

```diff
@@ -37,8 +207,13 @@ func (adminVdc *AdminVdc) Refresh() error {
 	// elements in slices.
 	unmarshalledAdminVdc := &types.AdminVdc{}
 
-	_, err := adminVdc.client.ExecuteRequest(adminVdc.AdminVdc.HREF, http.MethodGet,
-		"", "error refreshing VDC: %s", nil, unmarshalledAdminVdc)
+	// We are executing below request with a specific API version in the header, because we want to retrieve the most
+	// available fields in AdminVdc which vCD provides, but also which our code understands. As we can't blindly use
+	// the latest version, we're limiting the highest used version to the one we support with
+	// the GetSpecificApiVersionOnCondition(...) function. Specifically, the API version 32 returns
+	// two additional fields: IncludeMemoryOverhead and IsElastic for Flex allocation
+	_, err := adminVdc.client.ExecuteRequestWithApiVersion(adminVdc.AdminVdc.HREF, http.MethodGet,
+		"", "error refreshing VDC: %s", nil, unmarshalledAdminVdc, adminVdc.client.GetSpecificApiVersionOnCondition(">= 32.0", "32.0"))
```

Comment:
Can't we use `GetVDCByHref`  here?

Reply:
recepients are different. Refresh works with adminVDC and `GetVDCByHref` is for `adminOrg`